### ISSUE TITLE
validates_start_with_lowercase_after_prefix variant

### ISSUE THREAD

**Issue by donileo** [realm/SwiftLint#3306]
When using the following config which implies `validates_start_with_lowercase: true`:

```yml
identifier_name:
  allowed_symbols: ["_"]
```

The following code generates a `identifier_name` violation: 
 
```swift
let _myVar: String?
```

Swiftlint output : 
```
Identifier Name Violation: Variable name should start with a lowercase character: '_myVar' (identifier_name)
```

Based on what is occurring I think `validates_start_with_lowercase` is correctly doing what it says it does but not what I want it to do. What would be nice is if var names can be started with `_` but validations are still done to ensure that everything after the `_` starts with an alphanumeric character as `validates_start_with_lowercase` does now.

Maybe what would be good is a `validates_start_with_lowercase_after_prefix` rule with "_" in the list of prefixes.

The use case is that in swift theres a lot of the following and it would get tedious to have to give `_myVar` a different name while keeping its meaning:

```swift
let _myVar: String?

// Some _myVar init code here

guard let myVar = _myVar else { // More code here }
```
### Environment

* SwiftLint version 0.40.0
* Installation method: Homebrew

**Comment by jpsim** [comment]
Fixed in #3403
